homeassistant/components/feedreader/coordinator.py

```diff
@@ -0,0 +1,247 @@
+        # feedparser lib is trying a less strict parsing approach.
+        # If an error is detected here, log warning message but continue
+        # processing the feed entries if present.
+        if self._feed.bozo != 0:
+            _LOGGER.warning(
+                "Possible issue parsing feed %s: %s",
+                self._url,
+                self._feed.bozo_exception,
+            )
+        # Using etag and modified, if there's no new data available,
+        # the entries list will be empty
+        _LOGGER.debug(
+            "%s entri(es) available in feed %s",
+            len(self._feed.entries),
+            self._url,
+        )
+        if not self._feed.entries:
+            self._log_no_entries()
+            return None
+
+        self._filter_entries()
+        self._publish_new_entries()
+
+        _LOGGER.debug("Fetch from feed %s completed", self._url)
+
+        if (
+            self._has_published_parsed or self._has_updated_parsed
+        ) and self._last_entry_timestamp:
+            return self._last_entry_timestamp
+
+        return None
+
+    def _filter_entries(self) -> None:
+        """Filter the entries provided and return the ones to keep."""
+        assert self._feed is not None
+        if len(self._feed.entries) > self._max_entries:
+            _LOGGER.debug(
+                "Processing only the first %s entries in feed %s",
+                self._max_entries,
+                self._url,
+            )
+            self._feed.entries = self._feed.entries[0 : self._max_entries]
+
+    def _update_and_fire_entry(self, entry: feedparser.FeedParserDict) -> None:
+        """Update last_entry_timestamp and fire entry."""
+        # Check if the entry has a updated or published date.
+        # Start from a updated date because generally `updated` > `published`.
+        if "updated_parsed" in entry and entry.updated_parsed:
+            # We are lucky, `updated_parsed` data available, let's make use of
+            # it to publish only new available entries since the last run
+            self._has_updated_parsed = True
+            self._last_entry_timestamp = max(
+                entry.updated_parsed, self._last_entry_timestamp
+            )
+        elif "published_parsed" in entry and entry.published_parsed:
+            # We are lucky, `published_parsed` data available, let's make use of
+            # it to publish only new available entries since the last run
+            self._has_published_parsed = True
+            self._last_entry_timestamp = max(
+                entry.published_parsed, self._last_entry_timestamp
+            )
+        else:
+            self._has_updated_parsed = False
+            self._has_published_parsed = False
+            _LOGGER.debug(
+                "No updated_parsed or published_parsed info available for entry %s",
+                entry,
+            )
+        entry.update({"feed_url": self._url})
```

Comment:
```suggestion
        """Fire entry and return time_stamp of entry if provided."""
        # Check if the entry has a updated or published date.
        # Start from a updated date because generally `updated` > `published`.
        if time_stamp := entry.get("updated_parsed") or entry.get("published_parsed"):
            self._last_entry_timestamp = time_stamp
        else:
            _LOGGER.debug(
                "No updated_parsed or published_parsed info available for entry %s",
                entry,
            )
        entry["feed_url"] = self._url
```

The variables `self._has_updated_parsed` and `self._has_published_parsed` will set only by the last entry and I think they can be ommited

Reply:
4ba3d68776c989c947fe3af0b77c920bc139173a